pkg/edgeview/src/tcp.go

```diff
@@ -389,6 +432,44 @@ func setAndStartProxyTCP(opt string) {
 	}
 }
 
+// need to modify the kubeconfig file for mapping the edgeview local port for kubectl
+func genKubeConfigFile(kubeport int) (string, error) {
+	// this kubeYamlFile is user.yaml, which is not the kubeconfig file
+	// it only identify the user as 'debugging-user', who has only the kubernetes
+	// 'get', 'list' and 'watch' permissions
+	// thus the edgeview tcp/kube user can not change the kubernetes configuration
+	yamlBytes, err := os.ReadFile(kubeYamlFile)
+	if err != nil {
+		err = fmt.Errorf("tcp kube can not read/process kubeConfig yaml file. error %v\n", err)
+		return "", err
+	}
+
+	yamlString := string(yamlBytes)
+	pattern := regexp.MustCompile(`https://[\d\.]+:6443`)
+	newString := "https://localhost:" + strconv.Itoa(kubeport)
+
+	yamlString = pattern.ReplaceAllString(yamlString, newString)
+
+	err = os.WriteFile(kubeServerFile, []byte(yamlString), 0644)
+	if err != nil {
+		log.Errorf("genKubeConfigFile: write file error %v", err)
```

Comment:
I am confused when to use `log.Errorf` and when to use `fmt.Printf`; here `log.Errorf` is used, but https://github.com/lf-edge/eve/pull/3844/files#diff-32994ab68a36c3aa6e6e049881610984b70df1114fc0bf9f7ec39701d8471228R469 uses `fmt.Printf`

Reply:
@christoph-zededa this log.x used here, is to really do the logging for the device, when edgeview does fmt.Printf(), it actually uses the stdout to pipe into the data for the client side on the laptop. So some errors we want the client user user to know, mainly not allowed or policy does not match or something.

Reply:
So `fmt` is used to show the user directly the error message?
Shouldn't it then be here `fmt.Printf` as this directly impacts if this will work for the user or not, while for the other one it should be `log.Errorf` as it is not so bad if just a file cannot be deleted.

Reply:
yes, we are logging for the system here, but the err will burble up where the summarized message is sent back,
```
		kubefileName, err := genKubeConfigFile(kubeport)
		if err != nil {
			fmt.Printf("tcp kube copy file error: %v\n", err)
			return
		}
```

Reply:
Okay, thx for the explanation!